chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -304,15 +304,16 @@
       current_url = nil
       current_priority = nil
 
-      out = `LANG=C zypper --non-interactive repos #{name} 2> /dev/null`
-      out.split("\n").each do |line|
-        attribute, value = line.split(":", 2).map(&:strip)
-        next if value.nil?
-        if attribute == "URI"
-          current_url = value
-        elsif attribute == "Priority"
-          current_priority = value.to_i
-        end
+      out = REXML::Document.new(
+        `LANG=C zypper --non-interactive --xmlout repos Cloud 2> /dev/null`
+      ).root.elements
+
+      current_priority = unless out["repo-list/repo"].attributes["priority"].nil?
```

Comment:
I think this check can be dropped, as we can rely on the fact that there is always a url and a priority set for a repository. that would also make hound happy